music21/opensheetmusicdisplay/osmd.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+#-------------------------------------------------------------------------------
+# Name:         opensheetmusicdisplay/osmd.py
+# Purpose:      music21 classes for displaying music21 scores in IPython notebooks
+#
+# Authors:      Sven Hollowell
+#
+# Copyright:    Copyright © 2012-14 Michael Scott Cuthbert and the music21 Project
+# License:      LGPL or BSD, see license.txt
+#-------------------------------------------------------------------------------
+'''
+Registers the .show('osmd') converter for use in IPython browser
+See OSMD project page here: https://github.com/opensheetmusicdisplay/opensheetmusicdisplay
+'''
+import unittest
+from music21 import exceptions21
+
+
+class OpenSheetMusicDisplayException(exceptions21.Music21Exception):
+    pass
+
+hasInstalledIPython = False
+try:
+    # raise ImportError('for testing')
+    from IPython.core.display import display, HTML, Javascript
+    hasInstalledIPython = True
```

Comment:
This still will import upon load.  Look at base.py and graph.utilities.  The way to check for installation without importing is:

```python
    loader = importlib.util.find_spec('IPython.core.display')
    if loader is None:
        hasInstalledIPython = False
    else:
        hasInstalledIPython = True
```